crates/scroll/node/src/builder/network.rs

```diff
@@ -79,33 +135,192 @@ where
 pub type ScrollNetworkPrimitives =
     BasicNetworkPrimitives<ScrollPrimitives, scroll_alloy_consensus::ScrollPooledTransaction>;
 
-/// An implementation of a [`HeaderTransform`] for Scroll.
+/// The correct signer address for Scroll mainnet.
+const SCROLL_MAINNET_SIGNER: Address = address!("0xD83C4892BB5aA241B63d8C4C134920111E142A20");
+const SCROLL_SEPOLIA_SIGNER: Address = address!("0x687E0E85AD67ff71aC134CF61b65905b58Ab43b2");
+
+/// An implementation of a [`HeaderTransform`] for downloaded headers for Scroll.
 #[derive(Debug, Clone)]
 #[non_exhaustive]
 pub struct ScrollHeaderTransform<ChainSpec> {
     chain_spec: ChainSpec,
+    db: Arc<Database>,
 }
 
-impl<ChainSpec: ScrollHardforks + Debug + Send + Sync + 'static> ScrollHeaderTransform<ChainSpec> {
+impl<ChainSpec: EthChainSpec + ScrollHardforks + Debug + Send + Sync + 'static>
+    ScrollHeaderTransform<ChainSpec>
+{
     /// Returns a new instance of the [`ScrollHeaderTransform`] from the provider chain spec.
-    pub const fn new(chain_spec: ChainSpec) -> Self {
-        Self { chain_spec }
+    pub const fn new(chain_spec: ChainSpec, db: Arc<Database>) -> Self {
+        Self { chain_spec, db }
     }
 
     /// Returns a new [`ScrollHeaderTransform`] as a [`HeaderTransform`] trait object.
-    pub fn boxed<H: BlockHeader>(chain_spec: ChainSpec) -> Box<dyn HeaderTransform<H>> {
-        Box::new(Self { chain_spec })
+    pub fn boxed<H: BlockHeader>(
+        chain_spec: ChainSpec,
+        db: Arc<Database>,
+    ) -> Box<dyn HeaderTransform<H>> {
+        Box::new(Self { chain_spec, db })
+    }
+}
+
+impl<H: BlockHeader, ChainSpec: EthChainSpec + ScrollHardforks + Debug + Send + Sync>
+    HeaderTransform<H> for ScrollHeaderTransform<ChainSpec>
+{
+    fn map(&self, mut header: H) -> H {
+        if !self.chain_spec.is_euclid_v2_active_at_timestamp(header.timestamp()) {
+            return header;
+        }
+        // clear the extra data field.
+        // *header.extra_data_mut() = Default::default()
+
+        // TODO: remove this once we deprecated l2geth
+        // Validate and process signature
+        let authorized_signer = match self.chain_spec.chain().named() {
+            Some(NamedChain::Scroll) => Some(SCROLL_MAINNET_SIGNER),
+            Some(NamedChain::ScrollSepolia) => Some(SCROLL_SEPOLIA_SIGNER),
+            _ => None,
+        };
+
+        if let Err(err) =
+            self.validate_and_store_signature(&mut header, authorized_signer)
+        {
+            debug!(
+                target: "scroll::network::response_header_transform",
+                "Header signature persistence failed, header hash: {:?}, error: {}",
+                header.hash_slow(), err
+            );
+            return H::default();
```

Comment:
There may be circumstances where we have to change the signer key or no longer include the signature in the `extra_data` field. We still want l2reth to operate correctly in these circumstances. As such I don't think we should be returning the default header value. Instead we should just return the original header without modification.